openstack/nova-hypervisor-agents/bin/nova-compute-init

```diff
@@ -35,9 +29,14 @@ def _get_hypervisor():
     except (novaexceptions.NotFound, IndexError):
         return None
 
+
 compute_id = pathlib.Path("/var/lib/nova/compute_id")
 
 if not compute_id.exists():
     hypervisor = _get_hypervisor()
     if hypervisor:
-        pathlib.Path("/var/lib/nova/compute_id").write_text(hypervisor.id)
+        compute_id.write_text(hypervisor.id)
+    else:
+        node = platform.node().split(".", 1)[0]
+        hash = hashlib.sha1(node.encode(), usedforsecurity=False).hexdigest()
```

Comment:
sha1 is yielding 160 bit (20byte), which are 40 bytes hexadecimal string encoded.
uuidv4 is 128 bit (16 byte), 32 bytes payload with 4 bytes of dashes  = 36 bytes.

Afaik, nova-compute expects a valid uuid.

how about:

```
>>> compute_id = uuid.UUID(hashlib.md5('test'.encode(), usedforsecurity=False).hexdigest())
>>> hash = hashlib.md5('test'.encode(), usedforsecurity=False).hexdigest()
>>> compute_id = uuid.UUID(hash)
>>> compute_id.hex == hash
True
>>> str(compute_id)
'098f6bcd-4621-d373-cade-4e832627b4f6'
```

Reply:
I like the latter the most... It seems most standard way to get a uuid from a string.